From ea0059cf3f186901dea771e6e993db1eecbaeedb Mon Sep 17 00:00:00 2001
From: "nengwen.chen" <nengwen.chen@amlogic.com>
Date: Sun, 18 Feb 2024 16:15:05 +0800
Subject: [PATCH] dvb-core: add support SYS_DVBC_ANNEX_B and C in dvbv3 [1/1]

PD#SWPL-156400
PD#TV-109289

Problem:
SET_FRONTEND fail when use SYS_DVBC_ANNEX_B.

Solution:
add support SYS_DVBC_ANNEX_B and C in dvbv3.

Verify:
T5D.

Change-Id: I48f5901d37c05b1cc9eaa43e63af7e49361e93b5
Signed-off-by: nengwen.chen <nengwen.chen@amlogic.com>
---
 drivers/media/dvb-core/dvb_frontend.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/media/dvb-core/dvb_frontend.c b/drivers/media/dvb-core/dvb_frontend.c
index 59b4a92cc537e..144a1f27282b4 100644
--- a/drivers/media/dvb-core/dvb_frontend.c
+++ b/drivers/media/dvb-core/dvb_frontend.c
@@ -1683,20 +1683,17 @@ static int dtv_set_frontend(struct dvb_frontend *fe);

 static bool is_dvbv3_delsys(u32 delsys)
 {
-	bool status = false;
-
 #ifdef CONFIG_AMLOGIC_DVB_COMPAT
-	status = (delsys == SYS_DVBT) || (delsys == SYS_DVBC_ANNEX_A) ||
+	return (delsys == SYS_DVBT) || (delsys == SYS_DVBC_ANNEX_A) ||
+		 (delsys == SYS_DVBC_ANNEX_B) || (delsys == SYS_DVBC_ANNEX_C) ||
 		 (delsys == SYS_DVBS) || (delsys == SYS_ATSC) ||
 		 (delsys == SYS_DTMB) || (delsys == SYS_DVBS2) ||
 		 (delsys == SYS_DVBT2) || (delsys == SYS_ISDBC) ||
 		 (delsys == SYS_ISDBT) || (delsys == SYS_ISDBS);
 #else
-	status = (delsys == SYS_DVBT) || (delsys == SYS_DVBC_ANNEX_A) ||
+	return (delsys == SYS_DVBT) || (delsys == SYS_DVBC_ANNEX_A) ||
 		 (delsys == SYS_DVBS) || (delsys == SYS_ATSC);
 #endif
-
-	return status;
 }

 /**
--
2.42.0

